homeassistant/__main__.py

```diff
@@ -127,16 +188,19 @@ def main():
     config_dir = os.path.join(os.getcwd(), args.config)
     config_path = ensure_config_path(config_dir)
 
+    setup_daemon(args.pid_file, args.daemon)
+
     if args.demo_mode:
         from homeassistant.components import http, demo
 
         # Demo mode only requires http and demo components.
-        hass = bootstrap.from_config_dict({
-            http.DOMAIN: {},
-            demo.DOMAIN: {}
-        })
+        hass = bootstrap.from_config_dict(
+            {http.DOMAIN: {}, demo.DOMAIN: {}},
+            args=args)
```

Comment:
Please only sent minimum required information through. In this case `daemon=True/False`, `verbose=True/False`. That the data came from parsing args is an implementation detail of `__main__.py`.